tools/packgen/src/PackGen.cpp

```diff
@@ -92,29 +92,33 @@ int PackGen::RunPackGen(int argc, char *argv[]) {
     return 1;
   }
 
-  // Check CMake
-  result = generator.ExecCommand("cmake --version");
-  if (result.second) {
-    cerr << "packgen error: CMake was not found" << endl;
-    return 1;
-  }
+  // Avoid CMake calls when packs to be generated don't contain components and verbose mode is disabled
+  if (!generator.m_noComponents || generator.m_verbose) {
 
-  // Create CMake File API query
-  if (!generator.CreateQuery()) {
-    cerr << "packgen error: query creation failed" << endl;
-    return 1;
-  }
+    // Check CMake
+    result = generator.ExecCommand("cmake --version");
+    if (result.second) {
+      cerr << "packgen error: CMake was not found" << endl;
+      return 1;
+    }
 
-  // Parse CMake File API reply
-  if (!generator.ParseTarget()) {
-    cerr << "packgen error: target parsing failed" << endl;
-    return 1;
-  }
+    // Create CMake File API query
+    if (!generator.CreateQuery()) {
+      cerr << "packgen error: query creation failed" << endl;
+      return 1;
+    }
 
-  // Elaborate component contents
-  if (!generator.CreateComponents()) {
-    cerr << "packgen error: component generation failed" << endl;
-    return 1;
+    // Parse CMake File API reply
+    if (!generator.ParseTarget()) {
+      cerr << "packgen error: target parsing failed" << endl;
```

Comment:
Do we need to think about Error Numbers and adding more detailed documentation soon?
I like your brief error message but would prefer a bit more detail e.g. the name of the CMake target or maybe even the cmake command?
e.g.
'parsing of CMake file API response for target %s failed'

Reply:
I believe we should first consolidate the errlog library or to find a replacement before doing changes in this field.
The error message above is not valid anymore, ParseTarget prints warnings with relevant information inside the function body itself. I will clean it.